docs/advanced/misc.rst

```diff
@@ -304,3 +304,31 @@ the default settings are restored to prevent unwanted side effects.
 
 .. [#f4] http://www.sphinx-doc.org
 .. [#f5] http://github.com/pybind/python_example
+
+.. _avoid-cpp-types-in-docstrings:
+
+Avoid C++ types in docstrings
```

Comment:
Very minors: maybe "Avoiding ...", since this is on how to avoid those types, rather than "don't do this"?